File: app/src/main/kotlin/com/stroopoverload/domain/GameResult.kt
package com.stroopoverload.domain

data class GameResult(
    val finalScore: Int,
    val correctHits: Int,
    val totalRounds: Int,
    val survivalMs: Long,
    val previousHighScore: Int,
) {
    val accuracy: Int get() =
        if (totalRounds == 0) 0 else ((correctHits.toFloat() / totalRounds) * 100).toInt()

    val xpEarned: Int get() = (survivalMs / 1000 * 10).toInt()

    val isNewHighScore: Boolean get() = finalScore > previousHighScore
}

File: lib/domain/game_result.dart
class GameResult {
  const GameResult({
    required this.finalScore,
    required this.correctHits,
    required this.totalRounds,
    required this.survivalSeconds,
    required this.previousHighScore,
  });

  final int finalScore;
  final int correctHits;
  final int totalRounds;
  final double survivalSeconds;
  final int previousHighScore;

  int get accuracy =>
      totalRounds == 0 ? 0 : ((correctHits / totalRounds) * 100).round();

  int get xpEarned => (survivalSeconds * 10).round();

  bool get isNewHighScore => finalScore > previousHighScore;
}
